cllc-public-app/Controllers/PaymentController.cs

```diff
@@ -867,6 +867,170 @@ public async Task<IActionResult> VerifyLicenceFeePaymentStatus(string id, string
 
             return new JsonResult(response);
         }
+        
+        public static async Task<PaymentResult> GetPaymentStatus(MicrosoftDynamicsCRMadoxioApplication application, string invoiceType, IDynamicsClient dynamicsClient, Serilog.ILogger logger, IBCEPService bcep)
+        {
+            const string primary = "primary";
+            const string secondary = "secondary";
+
+            if (invoiceType != primary && invoiceType != secondary)
+            {
+                throw new Exception($"Invalid invoiceType specified: {invoiceType}");
+            }
+            if (application == null) throw new ArgumentNullException(nameof(application));
+            if (logger == null) throw new ArgumentNullException(nameof(logger));

```

Comment:
no need to fail if logger is not provided as you can simply use Log.

Reply:
This is static method. It can not use instance methods or properties